module/VuFind/src/VuFind/View/Helper/Root/UrlFactory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * ContentLoader helper factory.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2018.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <challber@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Interop\Container\ContainerInterface;
+use Zend\Router\RouteMatch;
+use Zend\ServiceManager\Factory\FactoryInterface;
+
+/**
+ * ContentLoader helper factory.
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Chris Hallberg <challber@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class UrlFactory implements FactoryInterface
+{
+    /**
+     * Create an object
+     *
+     * @param ContainerInterface $container     Service manager
+     * @param string             $requestedName Service being created
+     * @param null|array         $options       Extra options (optional)
+     *
+     * @return object
+     *
+     * @throws ServiceNotFoundException if unable to resolve the service.
+     * @throws ServiceNotCreatedException if an exception is raised when
+     * creating a service.
+     * @throws ContainerException if any other error occurs
+     */
+    public function __invoke(ContainerInterface $container, $requestedName,
+        array $options = null
+    ) {
+        $helper = new $requestedName($container->get('Request'));
```

Comment:
Also in the interest of avoiding redundancy, I wonder if you can restructure this to extend the parent factory. I realize that limits you since you can't inject the request through the constructor, but it might be worth adding a setter and injecting the request after the object has been built by the parent factory. You can add a safety check to throw an exception in addParameters if the dependency is missing. If you prefer to keep this approach, I can live with it... but the other option may be a little more future-proof if other details of the parent class change over time.

Reply:
Since the UrlFactory isn't a child of anything and the ViewHelperManagerFactory function this duplicates is private, this is, unfortunately, the simplest way to go.